File: decompiled/3.17.1/be/ehealth/businessconnector/chapterIV/utils/FolderTypeUtils.java
package be.ehealth.businessconnector.chapterIV.utils;

import be.fgov.ehealth.standards.kmehr.cd.v1.CDITEM;
import be.fgov.ehealth.standards.kmehr.cd.v1.CDITEMMAAvalues;
import be.fgov.ehealth.standards.kmehr.cd.v1.CDITEMschemes;
import be.fgov.ehealth.standards.kmehr.schema.v1.ContentType;
import be.fgov.ehealth.standards.kmehr.schema.v1.FolderType;
import be.fgov.ehealth.standards.kmehr.schema.v1.ItemType;
import be.fgov.ehealth.standards.kmehr.schema.v1.TransactionType;
import java.util.Iterator;
import java.util.List;
import org.joda.time.DateTime;

public final class FolderTypeUtils {
   public static DateTime retrieveConsultationStartDateOrAgreementStartDate(FolderType folder) {
      if (folder != null) {
         List<ItemType> items = ((TransactionType)folder.getTransactions().get(0)).getItem();
         Iterator i$ = items.iterator();

         while(i$.hasNext()) {
            ItemType itemType = (ItemType)i$.next();
            if (CDITEMMAAvalues.CONSULTATIONSTARTDATE.value().equals(((CDITEM)itemType.getCds().get(0)).getValue()) && ((CDITEM)itemType.getCds().get(0)).getS() == CDITEMschemes.CD_ITEM_MAA) {
               return ((ContentType)itemType.getContents().get(0)).getDate();
            }

            if (CDITEMMAAvalues.AGREEMENTSTARTDATE.value().equals(((CDITEM)itemType.getCds().get(0)).getValue()) && ((CDITEM)itemType.getCds().get(0)).getS() == CDITEMschemes.CD_ITEM_MAA) {
               return ((ContentType)itemType.getContents().get(0)).getDate();
            }
         }
      }

      return null;
   }
}

File: src/main/kotlin/org/taktik/connector/business/chapterIV/utils/FolderTypeUtils.kt
package org.taktik.connector.business.chapterIV.utils

import be.fgov.ehealth.standards.kmehr.cd.v1.CDITEM
import be.fgov.ehealth.standards.kmehr.cd.v1.CDITEMMAAvalues
import be.fgov.ehealth.standards.kmehr.cd.v1.CDITEMschemes
import be.fgov.ehealth.standards.kmehr.schema.v1.ContentType
import be.fgov.ehealth.standards.kmehr.schema.v1.FolderType
import be.fgov.ehealth.standards.kmehr.schema.v1.ItemType
import be.fgov.ehealth.standards.kmehr.schema.v1.TransactionType
import org.joda.time.DateTime

object FolderTypeUtils {
    fun retrieveConsultationStartDateOrAgreementStartDate(folder: FolderType?): DateTime? {
        if (folder != null) {
            (folder.transactions[0] as TransactionType).item.forEach {
                if (CDITEMMAAvalues.CONSULTATIONSTARTDATE.value() == (it.cds[0] as CDITEM).value && (it.cds[0] as CDITEM).s == CDITEMschemes.CD_ITEM_MAA) {
                    return (it.contents[0] as ContentType).date
                }
                if (CDITEMMAAvalues.AGREEMENTSTARTDATE.value() == (it.cds[0] as CDITEM).value && (it.cds[0] as CDITEM).s == CDITEMschemes.CD_ITEM_MAA) {
                    return (it.contents[0] as ContentType).date
                }
            }
        }
        return null
    }
}
